util/reciprocal_lattice/viewer.py

```diff
@@ -504,14 +522,24 @@ def DrawGL(self):
             self.draw_axis(self.rotation_axis, "phi")
         if self.beam_vector is not None and self.settings.show_beam_vector:
             self.draw_axis(self.beam_vector, "beam")
-        if self.recip_latt_vectors is not None and self.settings.show_reciprocal_cell:
-            for i, axes in enumerate(self.recip_latt_vectors):
-                if self.settings.experiment_ids:
-                    if i not in self.settings.experiment_ids:
-                        continue
-                j = (i + 1) % self.palette.size()
-                color = self.palette[j]
-                self.draw_cell(axes, color)
+
+        if self.settings.show_reciprocal_cell:
+            if self.settings.crystal_frame and self.recip_crystal_vectors:
+                for i, axes in enumerate(self.recip_crystal_vectors):
+                    if self.settings.experiment_ids:
+                        if i not in self.settings.experiment_ids:
+                            continue
+                    j = (i + 1) % self.palette.size()
+                    color = self.palette[j]
+                    self.draw_cell(axes, color)
+            elif self.recip_latt_vectors:
+                for i, axes in enumerate(self.recip_latt_vectors):
+                    if self.settings.experiment_ids:
+                        if i not in self.settings.experiment_ids:
+                            continue
+                    j = (i + 1) % self.palette.size()
+                    color = self.palette[j]
+                    self.draw_cell(axes, color)
```

Comment:
These conditional branches are basically duplicates. It would be cleaner to use the conditional up front to decide which list to loop over, then do the loop outside the conditional

Reply:
Refactored loop in fcf4d8944f785ad80d65faf2ce10cd97827e0b0a - not convinced it is actually clearer - after 736c374856398a8d8a6b60c6a91e96483ca7bddd maybe it is.